src/main/java/homework_2/random_chars_table/RandomCharsTable.java

```diff
@@ -0,0 +1,104 @@
+package homework_2.random_chars_table;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+public class RandomCharsTable {
+
+    public static final String ERROR = "Passed parameters should match the format [positive integer] [positive integer] [even|odd]";
+
+    private int length;
+    private int width;
+    private String strategy;
+
+    private String strategyEven = "Even letters -";
+    private String strategyOdd = "Odd letters -";
+
+    public void run() {
+
+        System.out.println("[Input example: 2 3 odd]");
+        System.out.println("Please, input 2 positive numbers and a strategy (even/odd)");
+
+        String inputFull = bufferedReaderReadConsole();
+        try {
+            parseInput(inputFull);
+            validateInputData(length, width, strategy);
+        } catch (IllegalArgumentException e) {
+            System.out.println(ERROR);
+            return;
+        }
+
+        char[][] table = new char[length][width];
+
+        for (int i = 0; i < length; i++) {
+            System.out.print("|");
+
+            for (int j = 0; j < width; j++) {
+                int x = (int) (Math.random() * 26 + 65);
+                table[i][j] = (char) x;
+                if (x % 2 == 0) {
+                    strategyEven = strategyEven + " " + table[i][j] + ",";
```

Comment:
Because of frequent concatenation it's better to use StringBuilder type for this variables (there would be no new object creation for each concatenation).
Also there is no need to announce this variables as global because you pass reference each time you call methods to operate on them.